pkg/util/vsphere/vm/cdrom.go

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) 2024 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package vm
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/vmware/govmomi/object"
+	vimtypes "github.com/vmware/govmomi/vim25/types"
+	ctrlclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	imgregv1a1 "github.com/vmware-tanzu/image-registry-operator-api/api/v1alpha1"
+
+	vmopv1 "github.com/vmware-tanzu/vm-operator/api/v1alpha3"
+	pkgctx "github.com/vmware-tanzu/vm-operator/pkg/context"
+)
+
+const (
+	vmiKind  = "VirtualMachineImage"
+	cvmiKind = "Cluster" + vmiKind
+
+	serverIDSuffix = "?serverId="
+)
+
+// UpdateCdromDeviceChanges updates the CD-ROM devices in the VM ConfigSpec to
+// reflect the desired state in the K8s VM Spec.
+func UpdateCdromDeviceChanges(
```

Comment:
I wonder if this should be a variation on the `EnsureAllDisksHaveControllers` function, ex. `EnsureAllCDROMsHaveControllers`. It would only need to consider IDE and AHCI (SATA) controllers, so it would be much simpler...